cpp/dolfinx/graph/KaHIP.cpp

```diff
@@ -66,113 +65,111 @@ graph::AdjacencyList<std::int32_t> dolfinx::graph::KaHIP::partition(
                       mode, &edgecut, part.data(), &mpi_comm);
   timer1.stop();
 
-  common::Timer timer2("Compute graph halo data (KaHIP)");
-
-  // Work out halo cells for current division of dual graph
-  const std::int32_t* xadj = adj_graph.offsets().data();
-  const unsigned long long* adjncy = adj_graph.array().data();
   const unsigned long long elm_begin = node_distribution[process_number];
   const unsigned long long elm_end = node_distribution[process_number + 1];
   const std::int32_t ncells = elm_end - elm_begin;
 
-  std::map<unsigned long long, std::set<std::int32_t>> halo_cell_to_remotes;
-  // local indexing "i"
-  for (int i = 0; i < ncells; i++)
+  // Create a map of local nodes to their additional destination processes,
+  // due to ghosting. If no ghosting, this will remain empty.
+  std::map<std::int32_t, std::set<std::int32_t>> local_node_to_dests;
+  if (ghosting)
   {
-    for (int j = 0; j < adj_graph.num_links(i); ++j)
+    common::Timer timer2("Compute graph halo data (KaHIP)");
+
+    // Work out halo cells for current division of dual graph
+    const std::int32_t* xadj = adj_graph.offsets().data();
```

Comment:
Can this be moved closer to where it's used?

Reply:
Move down and changed to `Eigen::Map`